README.md

```diff
@@ -23,30 +23,25 @@ Note the caveats: *experimental*, *simulator*, *research*, *potential*.
 
        poetry run demo
 
-## Programming patterns
+## Documentation
 
-The code makes use of the [simpy](https://simpy.readthedocs.io/en/latest/)
-discrete event simulation library. This means that functions representing
-processes are implemented as generators, so that the library can simulate
-timeouts and asynchronous communication (typically faster than real time).
+Design documentation is under the `doc/` directory:
 
-We use the convention of putting "(process)" in the doc comment of these
-functions. They either must use the `yield` construct, *or* return the
-result of calling another "(process)" function (not both).
+* [Programming patterns for use of simpy](doc/patterns.md).
 
-Objects that implement processes typically hold the `simpy.Environment` in
-an instance variable `self.env`.
+You can also generate API documentation by running `./gendoc.sh`.
```

Comment:
```suggestion
You can also generate API documentation by running `./gendoc.sh`. (You may need to run `poetry install` first to ensure the tools for doc generation are present.)
```

Inexperienced Python User experience report

This failed for me, and then I searched for a debian package named `pdoc`, then searched pypi to find it, then started to do a `pip` install it, then thought "wait, why not fix `gendoc.sh` so that it relies on poetry?", then looked at `gendoc.sh`, then realized I simply needed to run `poetry install` first.

Reply:
After `poetry install ; ./gendoc.sh` I verified the docs were rendered. However, I see `apidoc/` as an untracked file in git, so I request we add `apidoc/` to `.gitignore`.

Reply:
Tangential detail: I had to patch `pyproject.toml` to downgrade the python requirement as per https://github.com/zcash/simtfl/pull/19, but since the latter is already merged, I assume that should not block this PR.

Reply:
Done.